quark/subproject.py

```diff
@@ -952,7 +952,7 @@ def _download_and_hash_with_progress(self, req, dest: str) -> str:
         size = 0
         with open(dest, "wb") as output:
             for icon, chunk in zip(self.PROGRESS_ICONS, self._iter_chunks(req)):
-                if os.isatty(sys.stdout.isatty()):
+                if os.isatty(sys.stdout.fileno()):
```

Comment:
Nitpick: I would have gone for

```
                if sys.stdout.isatty():
```

besides being shorter and slightly less convoluted, it is a bit more general (a filelike object always has an `isatty()` method, that in the base class just returns `False`, while `fileno()` may throw if the implementation doesn't use OS fd objects; not that this matters a lot for `stdout`).

Reply:
This matches the same check done in `utils.print_msg()`. I would actually leave it like so in this commit and then introduce another commit that changes both uses to `sys.stdout.isatty()` so that we do the same everywhere.

Reply:
I added the aforementioned change in 56dced5